docs/compatibility.md

```diff
@@ -0,0 +1,101 @@
+
+## Semantic versioning
+
+Flow follows Semantic Versioning 2.0 (semver) for all components and projects. Each release version 
+of Flow has the form MAJOR.MINOR.PATCH. For example, Flow Protocol version 1.2.3 has MAJOR version 1, MINOR version 2, and PATCH version 3. Changes to each number have the following meaning:
+
+- **MAJOR:** Potentially backwards incompatible changes. Code and data that worked with a previous major release will not necessarily work with the new release.
+
+- **MINOR:** Backwards compatible features, speed improvements, etc. Code and data that worked with a previous minor release and which depends only on the public API will continue to work unchanged. For details on what is and is not the public API, see [Public APIs](#public-apis).
+
+- **PATCH:** Backwards compatible bug fixes.
+
+For example, release 1.0.0 introduced backwards incompatible changes from release 0.12.1. However, release 1.1.1 was backwards compatible with release 1.0.0.
+
+## Public APIs
+
+Each Flow component exposes a public API, loosely defined as the functionality that it exposes to other components or users.
+
+<table>
+    <thead>
+        <tr>
+            <th>Component</th>
+            <th>Subcomponent</th>
+            <th>Public API(s)</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan=5>Flow Protocol</td>
+            <td>Collection Node</td>
+            <td>
+              Ingestion Engine<br/>
+              Proposal Engine<br/>
+              Pusher Engine<br/>
+              Collection RPC API
+            </td>
+        </tr>
+        <tr>
+            <td>Consensus Node</td>
+            <td>
+              Ingestion Engine<br/>
+              Provider Engine<br/>
+              Matching Engine<br/>
+              Compliance Engine
+            </td>
+        </tr>
+        <tr>
+            <td>Execution Node</td>
+            <td>
+              Ingestion Engine<br/>
+              Provider Engine<br/>
+              Synchronization Engine<br/>
+              Flow Virtual Machine<br/>
+              Execution RPC API
+            </td>
+        </tr>
+        <tr>
+            <td>Verification Node</td>
+            <td>
+              Finder Engine<br/>
+              Matching Engine<br/>
+              Verifier Engine
+            </td>
+        </tr>
+        <tr>
+            <td>Access Node</td>
+            <td>Access RPC API</td>
+        </tr>
+        <tr>
+            <td>Flow Emulator</td>
+            <td></td>
+            <td>
+              Access RPC API<br/>
+              Flow Virtual Machine
+            </td>
+        </tr>
+        <tr>
+            <td>Cadence</td>
+            <td></td>
+            <td></td>
+        </tr>
+        <tr>
+            <td>Flow JavaScript SDK</td>
+            <td></td>
+            <td>Flow Client Library (FCL)</td>
+        </tr>
```

Comment:
@orodio @JeffreyDoyle Would you mind adding a brief description of the public API surfaces in the JS repo? It doesn't have to be overly descriptive here, even just a link to your README works.